### ISSUE TITLE
Spamming HA with entities

### ISSUE THREAD

**Issue by jchristianj** [jeroenterheerdt/HAsmartirrigation#441]
### What happened?

Each time the integration is reloaded, a new entitiy for each zone with suffix _n is created. 

### How to reproduce

Reload integration
[config_entry-smart_irrigation-c34d6b3034b69cc97400485090710263.json](https://github.com/user-attachments/files/16228548/config_entry-smart_irrigation-c34d6b3034b69cc97400485090710263.json)

### Which version are you running?

v2024.7.5

### Diagnostics file

- [X] I have attached a diagnostics file

**Comment by mistergyle** [comment]
duplicates https://github.com/jeroenterheerdt/HAsmartirrigation/issues/372

**Comment by jeroenterheerdt** [comment]
Duplicate of #372

**Comment by jeroenterheerdt** [comment]
uninstall, reinstall should fix this.

**Comment by jchristianj** [comment]
Unfortunately not. Completely removed the device, removed the integration, removed the HACS integration, rebooted, started from scratch with v2024.7.6, needed to configure everything (zones, modules, sensor group, weather API) new, but it still produces the *_n entities upon each reload.

**Comment by jeroenterheerdt** [comment]
OK, thanks. I will need to do some more digging then on why this happens. Thanks.

**Comment by jeroenterheerdt** [comment]
I tried, I tried, and I tried again but cannot reproduce this on any environment I have :(

**Comment by lukasz-burak** [comment]
I do have the same issue, entities just duplicate every integration refresh.
<img width="3108" alt="image" src="https://github.com/user-attachments/assets/1877087e-e0b6-4e30-bacb-dc13c6a7c1e5">

**Comment by MonsterEnergy-wtf** [comment]
Me2.
3 zones, 57 Entities. More to come 😁

**Comment by jeroenterheerdt** [comment]
I see two potential issues that might be causing it. The add this sensor activity gets called twice per zone and somehow the name generator adds the extra N all the time. Are you seeing the original zone and zones with _N added? Or just zone_Ns?

**Comment by jeroenterheerdt** [comment]
Come to think of it, was it working correctly for you before? If so, what version worked normally?

**Comment by MonsterEnergy-wtf** [comment]
I saw the regular one without the extension _n and a lot of same with the extension.
I endet up by removing the smart irrigation device ( not the integration) and added it again. The settings were kept and now everything seems to be fine. For now.

Not sure if this helps, but I played around with the experimental "Continuous updates for sensors". After my testing I realized that the enties were created. Maybe this is an side effect of the experimental function?

**Comment by jeroenterheerdt** [comment]
OK, maybe thanks for letting me know you found a solution. The others in this thread, can you let me know if the above helps for you as well?

**Comment by jchristianj** [comment]
Sorry, for me it is still not working and each reload of the integration doubles up my entities. I followed the steps to remove the smart irrigation device (which removes the integration as well) and re-adding it, with a HA restart in between. Unfortunately, all my settings were not kept and I had to reconfigure the integration, starting with the OWM Api Key and re-configuring all zones and modules.
